src/app/task/task-list/task-list.component.ts

```diff
@@ -36,18 +37,22 @@ export class TaskListComponent implements OnInit, OnDestroy {
   }
 
   updateTask(event: {id: string, task: Task}) {
-    this.taskService.updateTask(event.id, event.task)
-    .pipe(takeUntil(this.destroy$))
-    .subscribe(() => {
-      this.getTasks();
-    });
+    if (this.authService.isAdmin()) {
```

Comment:
instead of calling another request to the BE to get tasks, after updating a task in BE you could return a list of tasks with the updated task aswell 🙂